clpy/backend/function.pyx

```diff
@@ -87,22 +102,35 @@ cdef void _launch(clpy.backend.opencl.types.cl_kernel kernel, global_work_size,
                 if isinstance(a, clpy.core.core.Size_t):
                     if clpy.backend.opencl.utility.typeof_size() \
                             == 'uint':
-                        a = numpy.uint32(a.val)
+                        imm_uint = <cl_uint>a.val
+                        ptr = <size_t>&(imm_uint)
+                        size = cython.sizeof(cl_uint)
                     elif clpy.backend.opencl.utility.typeof_size() \
                             == 'ulong':
-                        a = numpy.uint64(a.val)
+                        imm_ulong = <cl_ulong>a.val
+                        ptr = <size_t>&(imm_ulong)
+                        size = cython.sizeof(cl_ulong)
                     else:
                         raise "api_sizeof_size is illegal"
                 elif isinstance(a, int):
-                    a = numpy.int_(a)
+                    imm_long = <cl_long>a
+                    ptr = <size_t>&(imm_long)
+                    size = cython.sizeof(cl_long)
                 elif isinstance(a, float):
-                    a = numpy.float_(a)
+                    imm_double = <cl_double>a
+                    ptr = <size_t>&(imm_double)
+                    size = cython.sizeof(cl_double)
                 elif isinstance(a, bool):
-                    a = numpy.bool_(a)
-
-                if core.numpy_scalar_type_set():
```

Comment:
In addition, I fixed this bug?
(L103 is always True)

Reply:
@ybsh Is it okay?

Reply:
@y1r It seems a bug and has not found because there aren't any test cases passing illegal type value.

@nsakabe-fixstars What do you suggest to do? Adding test cases? Upstream CuPy might include such test cases.

Reply:
@y1r As @LWisteria say, would you add a test / tests to cover these branches [here](https://github.com/fixstars/clpy/tree/clpy/tests/clpy_tests/core_tests)? When you do it, please leave a note (such as `NOTE(y1r): `) which leads to this discussion.

Reply:
@nsakabe-fixstars Adding tests (and also fixing bug) can be for another issue/PR? Is it should added by us or CuPy upstream?

Reply:
I have two options:
1. revert bugfix in this PR, add a new PR for bugfix and test, merge it into clpy, merge it into #281, merge #281 into clpy
2. add a new CuPy's PR for a test, merge it into CuPy, cherry-pick its change

I think option 1 is enough for us.

Reply:
There is no test in CuPy for this situation. Therefore, pulling upstream cannot resolve this issue.
As a summary, there are three options:
1. revert bugfix in this PR, add a new PR for bugfix and test, merge it into clpy, merge it into #281, merge #281 into clpy
2. add test in this PR
3. add a new CuPy's PR for a test, merge it into CuPy, cherry-pick its change

Reply:
@nsakabe-fixstars Which one is better for us?

Reply:
@y1r OK, I see your situation. Please take the option 1.